organization/models.py

```diff
@@ -2,105 +2,153 @@
+        verbose_name_plural = 'Internal groups'
 
-    members = models.ManyToManyField(
-        User,
-        blank=True,
-        related_name='internal_groups'  # The default django user model already has a `groups` related_name
-        # so we have to make a custom one
-    )
+    class Type(models.TextChoices):
+        """
+        Making use of https://docs.djangoproject.com/en/3.1/ref/models/fields/#enumeration-types
+        Denotes the internal group type, either a interest-group, e.g. KSG-iT, Påfyll etc. or an internal
+        grouping, e.g. Lyche bar og Edgar
+        """
+        INTEREST_GROUP = "interest-group"
+        INTERNAL_GROUP = "internal-group"
+
+    name = models.CharField(unique=True, max_length=32)
+    type = models.CharField(max_length=32, null=False, blank=False, choices=Type.choices)
+    description = models.TextField(max_length=2048, blank=True, null=True)
+
+    @property  # consider this not being a property seeing as its more "computationally" expensive than normal properties?
+    def active_members(self):
+        """
+        Returns all group positions membership with a FK to positions with a FK to this instance. Returns
+        only "active" memberships which is defined as memberships without an ending date
+        """
+        group_members = []
+        for position in self.positions.all():
+            group_members.extend(position.active_memberships.all())
+
+        group_members.sort(key=lambda x: x.user.get_full_name())
+        return group_members
+
+    @property
+    def active_members_count(self) -> int:
+        return len(self.active_members)
 
     def __str__(self):
         return "Group %s" % self.name
 
     def __repr__(self):
         return "Group(name=%s)" % self.name
 
-    class Meta:
-        default_related_name = 'groups'
-        verbose_name_plural = 'Internal groups'
+
+class InternalGroupPositionMembership(models.Model):
+    """
+    An intermediary model between a user and a InternalGroupPosition with additional information
+    regarding membership
+    """
+    date_joined = models.DateField(auto_now_add=True)
+    date_ended = models.DateField(
+        default=None,
+        null=True,
+        blank=True
+    )
+
+    user = models.ForeignKey("users.User", on_delete=models.CASCADE)
+    position = models.ForeignKey("organization.InternalGroupPosition", related_name="memberships",
+                                 on_delete=models.DO_NOTHING)
 
 
 class InternalGroupPosition(models.Model):
     """
     A position for an internal group, e.g. Hovmester
     """
-    STATUS = KSG_POSITIONS
 
-    name = StatusField(unique=True)
+    class Meta:
+        unique_together = ("name", "internal_group",)
 
-    description = models.CharField(max_length=1024, blank=True, null=True)
+    # feels natural to have the type here, like functionary or gang-member, but how do we deal with it as an
+    # interest group? Do we just have null=True and then not set it?
```

Comment:
I guess the enum can just be exhaustive here? I.e. let it be

* Functionary
* Gang-member
* Interest-group-member
* Active functionary pang
* Old functionary pang
* Active gang-member pang
* Old gang-member pang
* Hangaround